include/linux/ipts-companion.h

```diff
@@ -0,0 +1,22 @@
+#ifndef _INTEL_IPTS_COMPANION_H_
```

Comment:
this ifdef doesn't match the file name, and it's name does have a duplicate can these two be merged down to one?

Reply:
Yes the header guards are a total chaos in IPTS (everywhere, not just in this case). I would rather keep it that way now and fix them in a broader cleanup at some point though.

Regarding the duplicated filename: There is a second `ipts-companion.h` but I would like to keep them seperate because they do different things: The one in `include/` is everything that goes from the companion to IPTS. The other one is the opposite, it contains everything that goes from IPTS to the companion.

Reply:
I agree on the separation part, but the names being equal could cause some confusion. I don't have any suggestion for better names though.

Reply:
What we could do is remove the `ipts` prefix from all the files under `driver/misc/ipts`. You know they are for IPTS through the directory name anyways. Their names would be similar to the ones under `include/` but they wouldn't be the same.

That is totally out of scope for this pull request though imo.

Reply:
That might be a solution. Definitely agreed on the out-of-scope part.